### DISCUSSION TITLE
DLL import error

### DISCUSSION THREAD

**Discussion by alphaDeng** [IntelPython/dpctl#1177] [Q&A] [answered]
there are something wrong while i exectute the python script about dpctl....

i install dpctl from pypl，install it by using pip command.Then i tried to test it by execture a python script in documentation.And the python  interpreter return a error, the error is as follows:

Traceback (most recent call last):
  File "d:\software\programming\miniconda39\lib\site-packages\dpctl\__init__.py", line 34, in <module>
    from dpctl._sycl_context import SyclContext
  File "D:\software\programming\python\lib\site-packages\dpctl\__init__.py", line 34, in <module>
    from dpctl._sycl_context import SyclContext
ImportError: DLL load failed while importing _sycl_context: 找不到指定的模块。

it seem that .pyd file didn't work,but i don't know the reason.

is there any way to solve this problem?

**Comment by oleksandr-pavlyk** [comment, marked as answer]
The version of dpctl on PyPI is very much out of date (https://pypi.org/project/dpctl/#history) at 0.11.3

Failure of `dpctl._sycl_context` to import indicates failure to find and load dependent libraries, such as DPC++ Runtime library `libsycld.dll` which must exist on your computer (installed independently, i.e. from oneAPI installation) but must also be of the version compatible with the version of the DPC++ compiler used to build the `dpctl` binary package.

Could you please try installing `dpctl` using conda? Its advantage is that it would also install dependent libraries.

```sh
conda create -n dpctl -c intel dpctl
```

**Reply by alphaDeng** [reply]
it‘s work! It can find my CPU,and tell me more imformation about my CPU:

Get a list of all devices:

    Name            AMD Ryzen 7 4800H with Radeon Graphics
    Driver version  6.4.0.37
    Vendor          Intel(R) Corporation
    Filter string   opencl:cpu:0

=======================================

Get the list of only OpenCL devices:

    Name            AMD Ryzen 7 4800H with Radeon Graphics
    Driver version  6.4.0.37
    Vendor          Intel(R) Corporation
    Filter string   opencl:cpu:0

=======================================

Get all OpenCL CPU devices:

    Name            AMD Ryzen 7 4800H with Radeon Graphics
    Driver version  6.4.0.37
    Vendor          Intel(R) Corporation
    Filter string   opencl:cpu:0

=======================================

but it's possible that let dpctl find my GPU（RTX2060，CUDA11.4），maybe i should download something form conda?

Thank you!

**Reply by oleksandr-pavlyk** [reply]
`dpctl` depends on DPC++ runtime to find devices, and oneAPI DPC++ supports only Intel GPU devices out of the box. However, CodePlay provides oneAPI DPC++ [plugins](https://codeplay.com/portal/blogs/2022/12/16/bringing-nvidia-and-amd-support-to-oneapi) that enable support of NVidia and AMD GPUs. 

Once you install these, `dpctl` should start seeing your RTX2060 gpu. The `dpctl` installable through conda will not, however, be able to run kernels on this gpu. To do that one need to recompile `dpctl` after the relevant plugin is installed. 

See discission [#1124](https://github.com/IntelPython/dpctl/discussions/1124) for more details.
